### ISSUE TITLE
feature request: baml-cli dev --playground (or something similar)

### ISSUE THREAD

**Issue by sxlijin** [BoundaryML/baml#960]
For folks who don't use VSCode, we need a way to provide them with a decent developer experience.

The most likely path forward for this will be to allow users to spin up a local version of promptfiddle.com, backed by their local filesystem, something like

```
baml-cli dev --playground [port-number]
```

This will allow people to not have to download, install, and configure VSCode to edit their BAML code.

We've had a few requests from non-VSCode users about something along these lines: see #932 and #889

**Comment by zeionara** [comment]
Why do I need to spin up something besides LSP server at all? Why can't I rely on LSP server or at least just run some script to generate required files?

For example, when I develop `elixir` apps, I have an `LSP` server which I connect to from `neovim`, which recompiles the files on save. If I don't use `LSP`, I can run `mix compile` which generates required files. I would like the same experience.

**Comment by aaronvg** [comment]
Yep! I think setting up the lsp server we have for vscode separately can also work. We just need to look into how we can extract that out.

**Comment by sxlijin** [comment]
FYI, the language server now supports this! It will now print out a link to a `localhost` URL on startup, which if you're connecting to `baml-cli lsp` and treating it as an LSP, it will give you a BAML playground that you can work with<br><br>We do need to document this and clean up the interface a bit, but this is the mechanism that our Jetbrains and Zed extensions both rely on, so you can expect it to be stable and supported
